bin/build_utils

```diff
@@ -25,3 +25,24 @@ function gen_versions() {
     echo $version
   done
 }
+
+function git_tag_suffix() {
+  if [ $(is_tagged_commit) ]; then
+    echo "-$(git_tag)"
+  else
+    echo ""
+  fi
+}
+
+function is_tagged_commit() {
+  # fetching tags is required for git_description to work
+  git fetch --tags
+  git_description=$(git describe)
+
+  if [ "$git_description" != "v${VERSION}" ]; then
```

Comment:
we are trying to move away from using this logic in any pipelines, since it can lead to race conditions for builds

publish steps should always happen in the Jenkinsfile with the condition
`when { tag "v*" }`

in tag pipelines (ie when this condition is met), there is a `$TAG_NAME` environment variable that has the tag version. rather than do the logic you have here, you can just do
```suggestion
  if [[ ! -z "${TAG_NAME:-}" ]]; then
```

Reply:
for more info see: https://github.com/conjurinc/docs/blob/master/reference/jenkins.md#jenkins-pipeline-github-triggers